ql/ql/src/codeql_ql/performance/VarUnusedInDisjunctQuery.qll

```diff
@@ -0,0 +1,27 @@
+import ql
+
+/**
+ * Holds if we assume `t` is a small type, and
+ * variables of this type are therefore not an issue in cartesian products.
+ */
+predicate isSmallType(Type t) {
+  t.getName() = "string" // DataFlow::Configuration and the like
```

Comment:
Why are strings a small type? Small, here, means "low cardinality", right?

Edit: I see you've only moved a pre-existing predicate, but I'm still curious

Reply:
`string` is often a `Configuration` like class, like `DataFlow::Configuration`, and those are unused in a whole lot of predicates.

Reply:
🤷 ok! 👍